src/txdb.cpp

```diff
@@ -264,11 +280,39 @@ bool CCoinsViewDB::BatchWrite(CCoinsMap &mapCoins,
     
     for (CCswNullifiersMap::iterator it = cswNullifies.begin(); it != cswNullifies.end();) {
         const std::pair<uint256, libzendoomc::ScFieldElement>& position = it->first;
-        BatchWriteCswNullifier(batch, position.first, position.second, it->second.entered);
+        BatchWriteCswNullifier(batch, position.first, position.second, (it->second.flags == CCswNullifiersCacheEntry::FRESH));
```

Comment:
Pass as a third parameter `it->second` directly. Inside, do `switch(it->second.flag)` and process `DEFAULT` case as well.
Now, you will remove the entry even if the status was `DEFAULT`. See, `BatchSidechains` method,